libraries/Microsoft.Bot.Builder.Core.Extensions/MessageFactory.cs

```diff
@@ -145,16 +146,16 @@ public static IMessageActivity SuggestedActions(IList<string> actions, string te
         /// <returns>A message activity that contains the suggested actions.</returns>
         /// <exception cref="ArgumentNullException">
         /// <paramref name="cardActions"/> is <c>null</c>.</exception>
-        /// <seealso cref="SuggestedActions(IList{string}, string, string, string)"/>
-        public static IMessageActivity SuggestedActions(IList<CardAction> cardActions, string text = null, string ssml = null, string inputHint = null)
+        /// <seealso cref="SuggestedActions(IEnumerable{string}, string, string, string)"/>
+        public static IMessageActivity SuggestedActions(IEnumerable<CardAction> cardActions, string text = null, string ssml = null, string inputHint = null)
         {
             if (cardActions == null)
                 throw new ArgumentNullException(nameof(cardActions));
 
             IMessageActivity ma = Activity.CreateMessageActivity();
             SetTextAndSpeak(ma, text, ssml, inputHint);
 
-            ma.SuggestedActions = new SuggestedActions { Actions = cardActions };
+            ma.SuggestedActions = new SuggestedActions { Actions = cardActions?.ToList() };
```

Comment:
The "?" doesn't seem needed. We already know cardActions is not null (due to the null check), to ToList() will always work.

Reply:
agreed